Build failure: swayfx-unwrapped

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

Build `swayfx-unwrapped` (`x86_64-linux`) on current `master` (606996d74f6e2a12635d41c1bf58bfc7ea3bb5ec)

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/285791053

### Relevant log output

```console
[76/342] Compiling C object sway/sway.p/ipc-json.c.o
FAILED: sway/sway.p/ipc-json.c.o 
gcc -Isway/sway.p -Isway -I../sway -Iinclude -I../include -Iprotocols -I/nix/store/3aa3sws2gxg0jhhchdl2gn0kylxwl4wb-cairo-1.18.2-dev/include/cairo -I/nix/store/wy1fws8r7m3j9ivn8ha7vhyh9a40k1sa-freetype-2.13.3-dev/include/freetype2 -I/nix/store/wy1fws8r7m3j9ivn8ha7vhyh9a40k1sa-freetype-2.13.3-dev/include -I/nix/store/p43macm5bjxb066969xjwjhwns167drw-libdrm-2.4.124-dev/include -I/nix/store/p43macm5bjxb066969xjwjhwns167drw-libdrm-2.4.124-dev/include/libdrm -I/nix/store/x9dndiwzsifpxna1p5628qn5r514si8w-json-c-0.18-dev/include -I/nix/store/x9dndiwzsifpxna1p5628qn5r514si8w-json-c-0.18-dev/include/json-c -I/nix/store/v4kvbd1zsnyra01acv796khzb1j7426c-libevdev-1.13.3/include/libevdev-1.0/ -I/nix/store/y7m56imr1ccsy9nsl5522cdbl5gcmqlr-libinput-1.27.0-dev/include -I/nix/store/4idfxfbscpnnb52l1g0wvc49b3sabgfm-systemd-minimal-libs-257.2-dev/include -I/nix/store/jbhdy65rbla5zrkh5rmrsqnrbb5rczzr-glib-2.82.4-dev/include -I/nix/store/jbhdy65rbla5zrkh5rmrsqnrbb5rczzr-glib-2.82.4-dev/include/glib-2.0 -I/nix/store/qap69prdblhm5k4zkrqjk42zifyig2pb-glib-2.82.4/lib/glib-2.0/include -I/nix/store/b7hjjr6335dxgafc0r70qaqlldw7nflj-pango-1.55.5-dev/include/pango-1.0 -I/nix/store/vfbwnc5h21hhv2dac5cavh4wzidxklsi-harfbuzz-10.1.0-dev/include/harfbuzz -I/nix/store/3i082hldkgcvwl01k9ja1hgy3q20nsvk-pcre2-10.44-dev/include -I/nix/store/qvvh4qzs1dbny5gmf52gz35dcdsn39c1-pixman-0.44.2/include/pixman-1 -I/nix/store/vr1zy2jhjmszz52vcwkc717s9kssiqyf-scenefx-0.1/include -I/nix/store/7rm6qlssv445y64sxbg81gbhf8i4pmy2-wayland-1.23.1-dev/include -I/nix/store/1gsa77hq143k2ha2giw9jph5h0z4xra9-wlroots-0.17.4/include -I/nix/store/46ll5dgcqdx9x44cnqxm0lvh04i0cvm2-libxkbcommon-1.7.0-dev/include -I/nix/store/61x2m78ylwvw6cva8mpcyk1xqlfysrh6-libxcb-1.17.0-dev/include -I/nix/store/16y6hh95yww7ws61yy9vcg2mp4cf5rkc-xcb-util-wm-0.4.2-dev/include -I/nix/store/49w5nh9fmi8j2xalx777i0lcvakpwgp0-libglvnd-1.7.0-dev/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c11 -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wno-missing-braces -Wno-format-zero-length -Wundef -Wvla '-DSYSCONFDIR="/etc"' '-DNIX_SYSCONFDIR="//nix/store/nmjzskk9v3rrshghi60nxxxkl7zl4gdm-swayfx-unwrapped-0.4/etc"' '-DSWAY_VERSION="0.4"' '-DSWAY_ORIGINAL_VERSION="1.9.0"' -fmacro-prefix-map=../= -pthread -MD -MQ sway/sway.p/ipc-json.c.o -MF sway/sway.p/ipc-json.c.o.d -o sway/sway.p/ipc-json.c.o -c ../sway/ipc-json.c
../sway/ipc-json.c: In function 'describe_libinput_device':
../sway/ipc-json.c:974:17: error: enumeration value 'LIBINPUT_CONFIG_DRAG_LOCK_ENABLED_STICKY' not handled in switch [-Werror=switch]
  974 |                 switch (libinput_device_config_tap_get_drag_lock_enabled(device)) {
      |                 ^~~~~~
cc1: all warnings being treated as errors
[77/342] Compiling C object sway/sway.p/desktop_surface.c.o
[78/342] Compiling C object sway/sway.p/desktop_xdg_shell.c.o
[79/342] Compiling C object sway/sway.p/desktop_launcher.c.o
[80/342] Compiling C object sway/sway.p/desktop_layer_shell.c.o
[81/342] Compiling C object sway/sway.p/desktop_transaction.c.o
[82/342] Compiling C object sway/sway.p/input_seatop_down.c.o
[83/342] Compiling C object sway/sway.p/desktop_output.c.o
[84/342] Compiling C object sway/sway.p/input_input-manager.c.o
[85/342] Compiling C object sway/sway.p/input_seatop_move_floating.c.o
[86/342] Compiling C object sway/sway.p/input_seatop_move_tiling.c.o
[87/342] Compiling C object sway/sway.p/input_keyboard.c.o
[88/342] Compiling C object sway/sway.p/desktop_render.c.o
[89/342] Compiling C object sway/sway.p/input_cursor.c.o
[90/342] Compiling C object sway/sway.p/input_seatop_default.c.o
[91/342] Compiling C object sway/sway.p/input_seat.c.o
ninja: build stopped: subcommand failed.
```

### Additional context

PS this template could be simplified, https://hydra.nixos.org/job/nixpkgs/trunk/swayfx-unwrapped.x86_64-linux is a pretty easy URL to generate based on the attrname and platform.

### System metadata

-

### Notify maintainers

@RicArch97

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/nixos/tests/swayfx.nix
+++ b/nixos/tests/swayfx.nix
@@ -12,79 +12,77 @@ import ./make-test-python.nix (
     # Found 1 error in 1 file (checked 1 source file)
     skipTypeCheck = true;
 
-    nodes.machine =
-      { config, ... }:
-      {
-        # Automatically login on tty1 as a normal user:
-        imports = [ ./common/user-account.nix ];
-        services.getty.autologinUser = "alice";
+    nodes.machine = {
+      # Automatically login on tty1 as a normal user:
+      imports = [ ./common/user-account.nix ];
+      services.getty.autologinUser = "alice";
 
-        environment = {
-          # For glinfo and wayland-info:
-          systemPackages = with pkgs; [
-            mesa-demos
-            wayland-utils
-            alacritty
-          ];
-          # Use a fixed SWAYSOCK path (for swaymsg):
-          variables = {
-            "SWAYSOCK" = "/tmp/sway-ipc.sock";
-            # TODO: Investigate if we can get hardware acceleration to work (via
-            # virtio-gpu and Virgil). We currently have to use the Pixman software
-            # renderer since the GLES2 renderer doesn't work inside the VM (even
-            # with WLR_RENDERER_ALLOW_SOFTWARE):
-            # "WLR_RENDERER_ALLOW_SOFTWARE" = "1";
-            "WLR_RENDERER" = "pixman";
-          };
-          # For convenience:
-          shellAliases = {
-            test-x11 = "glinfo | tee /tmp/test-x11.out && touch /tmp/test-x11-exit-ok";
-            test-wayland = "wayland-info | tee /tmp/test-wayland.out && touch /tmp/test-wayland-exit-ok";
-          };
-
-          # To help with OCR:
-          etc."xdg/foot/foot.ini".text = lib.generators.toINI { } {
-            main = {
-              font = "inconsolata:size=14";
-            };
-            colors = rec {
-              foreground = "000000";
-              background = "ffffff";
-              regular2 = foreground;
-            };
-          };
-
-          etc."gpg-agent.conf".text = ''
-            pinentry-timeout 86400
-          '';
+      environment = {
+        # For glinfo and wayland-info:
+        systemPackages = with pkgs; [
+          mesa-demos
+          wayland-utils
+          alacritty
+        ];
+        # Use a fixed SWAYSOCK path (for swaymsg):
+        variables = {
+          "SWAYSOCK" = "/tmp/sway-ipc.sock";
+          # TODO: Investigate if we can get hardware acceleration to work (via
+          # virtio-gpu and Virgil). We currently have to use the Pixman software
+          # renderer since the GLES2 renderer doesn't work inside the VM (even
+          # with WLR_RENDERER_ALLOW_SOFTWARE):
+          # "WLR_RENDERER_ALLOW_SOFTWARE" = "1";
+          "WLR_RENDERER" = "pixman";
+        };
+        # For convenience:
+        shellAliases = {
+          test-x11 = "glinfo | tee /tmp/test-x11.out && touch /tmp/test-x11-exit-ok";
+          test-wayland = "wayland-info | tee /tmp/test-wayland.out && touch /tmp/test-wayland-exit-ok";
         };
 
-        fonts.packages = [ pkgs.inconsolata ];
+        # To help with OCR:
+        etc."xdg/foot/foot.ini".text = lib.generators.toINI { } {
+          main = {
+            font = "inconsolata:size=14";
+          };
+          colors = rec {
+            foreground = "000000";
+            background = "ffffff";
+            regular2 = foreground;
+          };
+        };
 
-        # Automatically configure and start Sway when logging in on tty1:
-        programs.bash.loginShellInit = ''
-          if [ "$(tty)" = "/dev/tty1" ]; then
-            set -e
-
-            mkdir -p ~/.config/sway
-            sed s/Mod4/Mod1/ /etc/sway/config > ~/.config/sway/config
-
-            sway --validate
-            sway && touch /tmp/sway-exit-ok
-          fi
+        etc."gpg-agent.conf".text = ''
+          pinentry-timeout 86400
         '';
-
-        programs.sway = {
-          enable = true;
-          package = pkgs.swayfx.override { isNixOS = true; };
-        };
-
-        # To test pinentry via gpg-agent:
-        programs.gnupg.agent.enable = true;
-
-        # Need to switch to a different GPU driver than the default one (-vga std) so that Sway can launch:
-        virtualisation.qemu.options = [ "-vga none -device virtio-gpu-pci" ];
       };
+
+      fonts.packages = [ pkgs.inconsolata ];
+
+      # Automatically configure and start Sway when logging in on tty1:
+      programs.bash.loginShellInit = ''
+        if [ "$(tty)" = "/dev/tty1" ]; then
+          set -e
+
+          mkdir -p ~/.config/sway
+          sed s/Mod4/Mod1/ /etc/sway/config > ~/.config/sway/config
+
+          sway --validate
+          sway && touch /tmp/sway-exit-ok
+        fi
+      '';
+
+      programs.sway = {
+        enable = true;
+        package = pkgs.swayfx.override { isNixOS = true; };
+      };
+
+      # To test pinentry via gpg-agent:
+      programs.gnupg.agent.enable = true;
+
+      # Need to switch to a different GPU driver than the default one (-vga std) so that Sway can launch:
+      virtualisation.qemu.options = [ "-vga none -device virtio-gpu-pci" ];
+    };
 
     testScript =
       { nodes, ... }:
@@ -194,7 +192,7 @@ import ./make-test-python.nix (
         swaymsg("exec swaylock")
         machine.wait_until_succeeds("pgrep -x swaylock")
         machine.sleep(3)
-        machine.send_chars("${nodes.machine.config.users.users.alice.password}")
+        machine.send_chars("${nodes.machine.users.users.alice.password}")
         machine.send_key("ret")
         machine.wait_until_fails("pgrep -x swaylock")
 

--- a/pkgs/by-name/sw/swayfx-unwrapped/package.nix
+++ b/pkgs/by-name/sw/swayfx-unwrapped/package.nix
@@ -33,6 +33,7 @@
   enableXWayland ? true,
   systemdSupport ? lib.meta.availableOn stdenv.hostPlatform systemd,
   trayEnabled ? systemdSupport,
+  fetchpatch2,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -56,6 +57,12 @@ stdenv.mkDerivation (finalAttrs: {
   patches =
     [
       ./load-configuration-from-etc.patch
+
+      (fetchpatch2 {
+        # fix missing switch statement for newer libinput
+        url = "https://github.com/swaywm/sway/pull/8470.patch?full_index=1";
+        hash = "sha256-UTZ2DNEsGi5RYrgZThHkYz3AnnIl/KxieinA1WUZRq4=";
+      })
 
       (substituteAll {
         src = ./fix-paths.patch;

